### DISCUSSION TITLE
expose `useItem` hook in extension sdk

### DISCUSSION THREAD

**Discussion by paullbn** [directus/directus#12758] [Feature Requests]
Hi,

After a quick discussion in the Discord's extension channel (https://discord.com/channels/725371605378924594/835244612356145167/964092666667204628) , i would like to know if there are any plans (or blocking points) to expose the `useItem` internal hook via the `@directus/extensions-sdk` to query a single item by id instead of using `useItems` with search/filter attributes (or recreating the hook localy in the extension directory).

Thanks,
Paul

**Comment by rijkvanzanten** [comment]
I don't think there's any major blockers! The main thing we have to consider when exposing more internals from the app to extensions is that we have to start treating those internals as breaking changes as well. Down for a PR if you want to move that useItem to shared 🙂

**Reply by paullbn** [reply]
Ok! I'll try to do it when I will be more comfortable with the codebase :) 
Thank you @rijkvanzanten

**Comment by adanielyan** [comment]
This would be great! I am trying to implement a custom endpoint for duplicating items, and I don't see how I can call the SaveAsCopy() function from useItem without manually copying the entire useItem hook into my extension.

**Comment by craigharman** [comment]
Did anything happen with this? I have a similar use case in my extension and can't see any way of doing it other than making my own API calls and validation.

**Comment by hanneskuettner** [comment]
@rijkvanzanten mind converting this into an issue so I can submit a PR that moves it over? Or has your opinion on it changed in the meantime?

**Reply by rijkvanzanten** [reply]
I've historically been very careful with exposing those bits, as they have a high chance of becoming breaking changes for extensions. That being said, we have a plan to properly version and communicate those changes in smaller sub-packages (like `@directus/composables`), so I think we're about ready to start opening things up more 🙂

**Comment by rijkvanzanten** [comment]
This feature request has been accepted and is queued to be implemented! You can follow along with the progress here: https://github.com/directus/directus/issues/18268
